# Data Processing

In [ ]:
from Agents import data_agent, badge_agent, report_agent, pre_game_report_agent
from openai import OpenAI
client = OpenAI(api_key= "")

profile_pth = "Data/Lakers/Anthony_Davis/Profile.json"
asb_pth = "Data/Lakers/Anthony_Davis/Advanced_Score_Boxes.csv"
os_pth = "Data/Lakers/Anthony_Davis/Overall_Shooting.csv"
grs_pth = "Data/Lakers/Anthony_Davis/General_Range_Shooting.csv"
ds_pth = "Data/Lakers/Anthony_Davis/Dribbles_Shooting.csv"
pt_pth = "Data/Lakers/Anthony_Davis/Pass_To.csv"
pf_pth = "Data/Lakers/Anthony_Davis/Pass_From.csv"
d_pth =  "Data/Lakers/Anthony_Davis/Tracking_Defense.csv"

# Data Processing
player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7 = data_agent.data_processing(profile_pth, asb_pth, os_pth, grs_pth, ds_pth, pt_pth, pf_pth, d_pth)

# Generate Player Badges

In [17]:
# Badge Generation
badge_user_input = badge_agent.create_badge_input(player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7)
badge_result = badge_agent.generate_badge(client, badge_user_input)
print(badge_result)

Defense Badge  
Inside Scorer Badge  
Rebounding Badge


# Generate Player Report

In [9]:
# Report Generation
report_user_input = report_agent.create_report_input(player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7)
report_result = report_agent.generate_report(client, report_user_input)
print(report_result)

**Pre-Game Scouting Report: Anthony Davis**

**Overview:**

- **Player's Profile:**  
  - Name: Anthony Davis
  - Number: 3
  - Team: Los Angeles Lakers
  - Position: Power Forward, Center
  - Height: 6 feet 10 inches (2.08 meters)
  - Weight: 253 pounds (115 kilograms)

- **Advanced Scoreboxes Average Stats (Last 12 Games):**  
  - PPG (Points Per Game): 29.2  
  - RPG (Rebounds Per Game): 11.9  
  - APG (Assists Per Game): 2.5  
  - SPG (Steals Per Game): 1.3  
  - BPG (Blocks Per Game): 2.1  
  - FG%: 56.1%  
  - 3P%: 42.9%  
  - FT%: 77.0%

- **Recent Performance Trends:**  
  Anthony Davis has been in excellent form, showcasing high shooting efficiency and strong defensive capabilities. His scoring has been consistent, particularly effective in the paint and mid-range areas. Although his performance shows a steady pace, there was a slight dip against Memphis indicating periodic variability. Overall, he is currently on a hot streak with his scoring efficiency and defensive impact.


# Generate Player Pre-Game Report

In [10]:
pre_game_report_user_input = pre_game_report_agent.create_pre_game_report_input(report_result,  badge_result)
pre_game_report_result = pre_game_report_agent.generate_pre_game_report(client, pre_game_report_user_input)

# Pre-Game Scouting Report

In [11]:
print(pre_game_report_result)

```markdown
<div style="text-align:center;">
  <span style="font-size: 32px; font-weight: bold;">Pre-Game Scouting Report</span>
</div>

<div margin: 20px 0;>
<img src="Lakers/Anthony_Davis/Portrait.jpg" alt="Portrait" width="520" height="380" style="border: 1px solid #ddd; border-radius: 5px; padding: 5px;">
</div>

<div margin: 20px 0;>
  <img src="Badges/Defense.PNG" alt="Defense Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Finisher.PNG" alt="Inside Scorer Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Rebound.PNG" alt="Rebounding Badge" width="92" height="104" style="margin: 0 5px;">
</div>

<div>
  <span style="font-size: 32px; font-weight: bold;">Anthony Davis</span><br>
  <span style="font-size: 24px;">Los Angeles Lakers | #3 | Power Forward, Center</span>
</div>

<div style="margin: 30px 0;">
  <table style="width: 100%; border-collapse: collapse; text-align: center;">
    <thead style="background-color">
      <tr>
      

<div style="text-align:center;">
  <span style="font-size: 32px; font-weight: bold;">Pre-Game Scouting Report</span>
</div>

<div margin: 20px 0;>
<img src="Data/Lakers/Anthony_Davis/Portrait.jpg" alt="Portrait" width="520" height="380" style="border: 1px solid #ddd; border-radius: 5px; padding: 5px;">
</div>

<div margin: 20px 0;>
  <img src="Badges/Defense.PNG" alt="Defense Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Finisher.PNG" alt="Finisher Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Rebound.PNG" alt="Rebounding Badge" width="92" height="104" style="margin: 0 5px;">
</div>

<div>
  <span style="font-size: 32px; font-weight: bold;">Anthony Davis</span><br>
  <span style="font-size: 24px;">Los Angeles Lakers | #3 | Power Forward/Center</span>
</div>

<div style="margin: 30px 0;">
  <table style="width: 100%; border-collapse: collapse; text-align: center;">
    <thead style="background-color">
      <tr>
        <th style="padding: 10px; border: 1px solid #ddd;">PPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">RPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">APG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">SPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">BPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">FG%</th>
        <th style="padding: 10px; border: 1px solid #ddd;">3P%</th>
        <th style="padding: 10px; border: 1px solid #ddd;">FT%</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td style="padding: 10px; border: 1px solid #ddd;">30.7</td>
        <td style="padding: 10px; border: 1px solid #ddd;">12.0</td>
        <td style="padding: 10px; border: 1px solid #ddd;">2.5</td>
        <td style="padding: 10px; border: 1px solid #ddd;">1.0</td>
        <td style="padding: 10px; border: 1px solid #ddd;">2.1</td>
        <td style="padding: 10px; border: 1px solid #ddd;">56.1%</td>
        <td style="padding: 10px; border: 1px solid #ddd;">42.9%</td>
        <td style="padding: 10px; border: 1px solid #ddd;">78.8%</td>
      </tr>
    </tbody>
  </table>
</div>

## Key Strengths:
- Excellent shooting efficiency with high FG% and 3P%.
- Dominant rebounder with consistent double-digit boards.
- Strong defender impacting opponents' scoring efficiency.

---

## Key Weaknesses:
- Vulnerable during pull-up shot scenarios.
- Higher defensive field goal percentage in the paint.
- Frequency of three-point attempts is relatively low.

---

## Offensive Strategy:
- Attack him in the paint to challenge his higher DFG%.
- Use pick-and-roll to create mismatches with perimeter guards.

---

## Defensive Strategy:
- Force contested mid-range shots.
- Deny easy catches and double in the post if necessary.